File: Programación Orientada a Objetos (PDOO)/Napakalaki(Ruby)/lib/bad_consequence.rb
#encoding: utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative "prize.rb"
require_relative "monster.rb"
require_relative "treasure_kind.rb"

module NapakalakiGame
class BadConsequence

  @@MAXTREASURES=10
  attr_accessor:text
  attr_accessor:levels
  attr_accessor:nVisibleTreasures
  attr_accessor:nHiddenTreasures
  attr_accessor:death
  attr_accessor:specificVisibleTreasures
  attr_accessor:specificHiddenTreasures
  
  
  def initialize(aText,someLevels)
    @text=aText  #txt
    @levels=someLevels  #lvls
  end
 
  def isEmpty()
  end
   
  def substractVisibleTreasures(t)
  end
 
  def substractHiddenTreasures(t)
  end
 
  def setnVisibleTreasures(nV)
  end
  
  def setnHiddenTreasures(nH)
  end
  
  def adjustToFitTreasureList(v,h)
  end

  def to_s
    "#{@text}\n Niveles perdidos: #{@levels}" 
  end
  
end
end

File: Programación Orientada a Objetos (PDOO)/Napakalaki(Java)/src/napakalaki/BadConsequence.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package napakalaki;
import java.util.ArrayList;         // ArrayList se encuentra en el paquete java.util
/**
 *
 * @author joseperoman
 */
public abstract class BadConsequence {
    //declarar constante MAXTREASURES (protected)
    protected static final int MAXTREASURES=10;
    protected String text;
    protected int levels;
  

    
    public BadConsequence(String txt, int lvl){
        text = txt;
        levels = lvl;    
    }
  
    
    public String getText(){
        return text;
    }
    public int getLevels(){
        return levels;
    }
   
    
    public abstract boolean isEmpty();
    
    public abstract void substractVisibleTreasure(Treasure t);
    public abstract void substractHiddenTreasure(Treasure t);
   
    
     public abstract BadConsequence adjustToFitTreasureList(ArrayList<Treasure> v, ArrayList<Treasure> h);
    
    
    public String toString(){
        return " \n\tTexto = " + text + " \n\tLevels = " + Integer.toString(levels);
    }
}